### PR TITLE
Fix/muwm 3796

### PR DESCRIPTION
**Pull request by fanglinfang**
https://jira.cac.washington.edu/browse/MUWM-3796

### CODE REVIEW

File: myuw/templates/handlebars/card/textbook.html [hunk 1]
```diff
@@ -25,7 +25,7 @@
 
         {{#unless no_book_assigned}}
             <div class="destination-arrow">
-                <a class="show_textbooks" href="/textbooks/{{year}},{{quarter}}{{#if summer_term}},{{toLowerCase summer_term}}{{/if}}" title="View textbooks" aria-label="View textbook details for this quarter" ><i role="presentation" class="fa fa-chevron-right"></i></a>
+                <a class="show_textbooks" id="show_term_textbooks" href="/textbooks/{{year}},{{quarter}}{{#if summer_term}},{{toLowerCase summer_term}}{{/if}}" title="View textbooks" aria-label="View textbook details for this quarter" ><i role="presentation" class="fa fa-chevron-right"></i></a>
```
Comment (vegitron) [inline myuw/templates/handlebars/card/textbook.html]: Is this going to be on the page multiple times, e.g. summer and autumn?  the ids should be unique.

File: myuw/test/views/textbooks.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 200)
+        self.assertIsNone(response.context['term'])
+        self.assertIsNone(response.context['textbook'])
+        self.assertEqual(response.context['year'], 2013)
+        self.assertEqual(response.context['quarter'], 'spring')
+
+        response = self.client.get(url + "/2013,spring/PHYS121AC")
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.context['term'], u'2013,spring')
```
Comment (vegitron) [inline myuw/test/views/textbooks.py]: Good, this is the sort of testing I was hoping to see.

File: myuw/test/views/textbooks.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        self.assertEqual(response.context['textbook'], u'PHYS121AC')
+
+    @skipIf(missing_url("myuw_textbooks_page"),
+            "myuw_pref_new_site urls not configured")
+    def test_textbook_future(self):
+        url = reverse("myuw_textbooks_page")
+        self.set_user("javerage")
+        response = self.client.get(url + "/2013,autumn")
```
Comment (vegitron) [inline myuw/test/views/textbooks.py]: Can this url be generated by reverse() w/ arguments?  The name might need to be updated in urls.py - i don't know what happens if 2 url patterns have the same name.

File: myuw/templates/handlebars/card/textbook.html (hunk 1 above)
Comment (fanglinfang) [inline myuw/templates/handlebars/card/textbook.html]: this is on a future quarter preview page, there is only one textbook card on the page.

File: myuw/test/views/textbooks.py (hunk 3 above)
Comment (fanglinfang) [inline myuw/test/views/textbooks.py]: i'd like to use the exact URL arguments come in to test out the pattern.

File: myuw/templates/handlebars/card/textbook.html (hunk 1 above)
Comment (vegitron) [inline myuw/templates/handlebars/card/textbook.html]: alright, and it looks like the landing page only shows the current term's textbooks.

File: myuw/test/views/textbooks.py (hunk 3 above)
Comment (vegitron) [inline myuw/test/views/textbooks.py]: i think that's fine, but (eventually) i would like to see the reverse tested and used, and maybe having tests that verify the patterns match what we think they should match split out

**Pull request merged**